nntrainer/opencl/opencl_context_manager.cpp

```diff
@@ -178,6 +178,7 @@ bool ContextManager::CreateDefaultGPUDevice() {
     ml_loge("fp16 (half) is not supported by device");
     return false;
   }
+  delete[] extensions;
```

Comment:
This may potentially result in a memory leak. when the `status` is not CL_SUCCESS, it will exit from a function without freeing.

Reply:
maybe we can add `delete[] extensions` before `return false` for both if statements.

Reply:
or use `std::vector<char> extensions(extension_size)`.